File: 1700. Number of Students Unable to Eat Lunch.cpp
//
// Created by leeyu on 2023/8/4.
//
#include <iostream>
using namespace std;

class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        while (sandwiches.size() != 0) {
            if (find(students.begin(), students.end(), sandwiches[0]) != students.end()) {
                if (sandwiches[0] == students[0]) {
                    sandwiches.erase(sandwiches.begin());
                    students.erase(students.begin());
                } else {
                    students.push_back(students[0]);
                    students.erase(students.begin());
                }
            } else {
                break;
            }
        }
        return students.size();
    }
};

File: 1700. Number of Students Unable to Eat Lunch.py
# 
# Created by leeyu on 2022/10/6.
# 
from typing import List

class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        while sandwiches:
            if sandwiches[0] not in students:
                break
            if students[0] == sandwiches[0]:
                del students[0]
                del sandwiches[0]
            else:
                students.append(students[0])
                del students[0]



        return len(students)
    
if __name__ == '__main__':
    S = Solution()
    num = S.countStudents([1,1,0,0], [0,1,0,1])
    print(num)